cmd/craned/app/manager.go

```diff
@@ -197,8 +203,27 @@ func initDataSources(mgr ctrl.Manager, opts *options.Options) (map[providers.Dat
 	return realtimeDataSources, historyDataSources, hybridDataSources
 }
 
-func initPredictorManager(opts *options.Options, realtimeDataSources map[providers.DataSourceType]providers.RealTime, historyDataSources map[providers.DataSourceType]providers.History) predictor.Manager {
-	return predictor.NewManager(realtimeDataSources, historyDataSources, predictor.DefaultPredictorsConfig(opts.AlgorithmModelConfig))
+func initPredictorManager(opts *options.Options, realtimeDataSources map[providers.DataSourceType]providers.RealTime, historyDataSources map[providers.DataSourceType]providers.History) (predictor.Manager, error) {
+	cpStoreType := checkpoint.StoreType(opts.CheckpointerStore)
+	var checkpointer checkpoint.Checkpointer
+	var err error
+	if opts.EnableCheckpointer {
+		switch cpStoreType {
+		case checkpoint.StoreTypeLocal:
+			checkpointer, err = checkpoint.InitCheckpointer(checkpoint.StoreType(opts.CheckpointerStore), opts.CheckpointerLocalConfig)
+			if err != nil {
```

Comment:
> should the InitCheckpointer be called exactly once? e.g., by sync.Once.Do

it is ok to call InitCheckpointer more than once, becasue the factory pattern, there is a map to store each storage checkpoint.  it will return if it already exists.

Reply:
LGTM